### ISSUE TITLE
No provider for "framework:phantomjs-shim"!

### ISSUE THREAD

**Issue by jpetitcolas** [tschaub/karma-phantomjs-shim#1]
I got an error following your README instructions:

> Running "karma:unit" (karma) task
> Warning: No provider for "framework:phantomjs-shim"! (Resolving: framework:phantomjs-shim) Use --force to continue.

Here is my configuration file:

``` js
module.exports = function (config) {
    'use strict';
    config.set({
        // ...
        frameworks: ['requirejs', 'jasmine', 'phantomjs-shim'],
        // ...
        plugins: ['karma-requirejs', 'karma-jasmine', 'karma-chrome-launcher', 'karma-phantomjs-launcher', 'karma-babel-preprocessor'],
        files: [
            // ...
        ],
        preprocessors: {
            'lib/**/*.js': 'babel'
        },
        babelPreprocessor: {
            options: {
                modules: "amd"
            }
        }
    });
};
```

And the `package.json`:

``` js
{
  "devDependencies": {
    // ...
    "karma": "~0.12.14",
    "karma-babel-preprocessor": "^4.0.1",
    "karma-chrome-launcher": "^0.1.4",
    "karma-jasmine": "~0.3.3",
    "karma-ng-html2js-preprocessor": "~0.1.0",
    "karma-ng-scenario": "~0.1.0",
    "karma-phantomjs-launcher": "^0.1.4",
    "karma-phantomjs-shim": "^1.0.0",
    "karma-requirejs": "~0.2.2",
  },
  // ...
}
```

The [Travis build](https://travis-ci.org/marmelab/ng-admin/builds/55920559) and [configuration](https://github.com/marmelab/ng-admin/blob/fix_tests/src/javascripts/test/karma.conf.js) are taken from [ng-admin](https://github.com/marmelab/ng-admin).

**Comment by amol-patil** [comment]
We are also seeing this issue

**Comment by jpetitcolas** [comment]
@amol-patil, here is a workaround we used: https://github.com/marmelab/ng-admin/commit/af4d303edfc1149861de0a1dea2fdd212afcebcc

**Comment by chriscollinsboxuk** [comment]
I came across the same issue.  I believe it's happening because you've defined the `plugins` section in your Karma config file.  Without it defined, Karma will load all NPM dependencies beginning with `karma-` automatically.  If you've defined it however, you'll also need to add a `'karma-phantomjs-shim'` entry to your plugins array.

This fixed the problem for me.

**Comment by curiouslychase** [comment]
:+1: to @chriscollinsboxuk suggestion, that was what worked for me too.

**Comment by riddla** [comment]
@chriscollinsboxuk`s suggestion helped me, too. This should be addressed in the README.

**Comment by tschaub** [comment]
I'd gladly accept a pull request that clarified the README.

**Comment by riddla** [comment]
I used parts of @chriscollinsboxuk`s suggestion text to complement the instructions.

**Comment by ccorcos** [comment]
@chriscollinsboxuk is right. This should probably be in the readme.

``` coffee
      karma:
        options:
          browsers: ['PhantomJS']
          frameworks: ['phantomjs-shim']
          plugins: ['karma-phantomjs-shim']
```

**Comment by tschaub** [comment]
Fixed with #3.  Published as 1.1.2.  Thanks @riddla.
